src/client/components/Buy.tsx

```diff
@@ -47,16 +54,43 @@ export const Buy: FunctionComponent<BuyProps> = ({ itemID, togglePaymentState })
         try {
             // Send the transaction to the network
             const txHash = await sendTransaction(tx, connection);
-            //NIKITA : VALIDATE payment here , this is set to true assuming all payments passsed
-            setPaid(true);
-            togglePaymentState(false);
+            setStatus(STATUS.Submitted);
         } catch (error) {
             console.error(error);
```

Comment:
It'll be nice to update this to show an error alert since you're validating transactions